File: python/040_house/house.py
phrases = [x + " that " + y for x, y in (("house", "Jack built."), ("malt", "lay in"), ("rat", "ate"), ("cat", "killed"), ("dog", "worried"), ("cow with the crumpled horn", "tossed"), ("maiden all forlorn", "milked"), ("man all tattered and torn", "kissed"), ("priest all shaven and shorn", "married"), ("rooster that crowed in the morn", "woke"), ("farmer sowing his corn", "kept"), ("horse and the hound and the horn", "belonged to"))]

def recite(start_verse, end_verse):
    return [recite_verse(number) for number in range(start_verse, end_verse + 1)]
    
def recite_verse(number):
    return ' the '.join(['This is'] + phrases[:number][::-1])

File: javascript/ex_084_house/house.js
export class House {
  static verse(n) {
    const phrases = [
      ['lay in', 'house that Jack built.'],
      ['ate', 'malt'],
      ['killed', 'rat'],
      ['worried', 'cat'],
      ['tossed', 'dog'],
      ['milked', 'cow with the crumpled horn'],
      ['kissed', 'maiden all forlorn'],
      ['married', 'man all tattered and torn'],
      ['woke', 'priest all shaven and shorn'],
      ['kept', 'rooster that crowed in the morn'],
      ['belonged to', 'farmer sowing his corn'],
      ['', 'horse and the hound and the horn']
    ];

    const lines = [];
    for (let i = n - 1; i >= 0; i--) {
      if (i === n - 1)
        lines.push(`This is the ${phrases[i][1]}`);
      else
        lines.push(`that ${phrases[i][0]} the ${phrases[i][1]}`);
    }
    return lines;
  }

  static verses(startVerse, endVerse) {
    const allVerses = [];
    for (let i = startVerse; i <= endVerse; i++) {
      allVerses.push(...this.verse(i));
      if (i !== endVerse)
        allVerses.push('');
    }
    return allVerses;
  }
}